kbase-extension/static/kbase/js/widgets/function_output/kbaseExpressionPairwiseCorrelation.js

```diff
@@ -53,86 +94,165 @@ define(['kbwidget', 'bootstrap', 'jquery', 'kbaseExpressionGenesetBaseWidget', '
             };
         },
 
-        buildWidget: function ($containerDiv) {
-            const self = this;
+        $renderDownloadControl: function ($heatmapContainer, label, controlType = 'button') {
+            const downloadHeatmap = () => {
+                // Extract the heatmap svg content
+                const file = new Blob([$heatmapContainer.find('svg')[0].outerHTML], {
+                    type: 'text',
+                });
+
+                // Create a donor link element which will be "clicked" to download the content
+                // extracted above.
+                const link = document.createElement('a');
+                const url = URL.createObjectURL(file);
+                link.href = url;
+                link.download = 'pairwise.svg';
+                document.body.appendChild(link);
+                link.click();
+
+                // Theoretically this ensures that the link and url are not needed any more.
+                // I don't know of documentation for this behavior, but it does seem to work.
+                // Note that there is nothing really to "download", as the heatmap content is
+                // already loaded in the browser.
+                setTimeout(() => {
```

Comment:
Is the `setTimeout` wrapper bit necessary? Is that something to do with the Blob lifecycle?

Reply:
Just added this to the comment:

I didn't find documentation for this, but I figure that the click is
not handled until the next turn of the wheel in the JS event loop. So the
timeout just lets the event loop handle the click, and timers are executed
after any pending events, so this should ensure that the link being clicked
is not removed until the link 'click' is handled.